Fix travis integration for netifaces and netsnmp modules.

stephen-soltesz opened the pull request:
As a security precaution, travis-ci maintains a whitelist of packages that can be installed for tests. Any package not on the whitelist cannot be installed for CI testing. The Ubuntu packages `python-netifaces` and `python-netsnmp` are not on the whitelist. Similarly, the pip packages for `netifaces` and `netsnmp` cannot build because their dependencies are not whitelisted either, e.g. libsnmp-dev.

Since these packages involve network discovery and system introspection (potential security risks), it is unlikely that travis would whitelist these packages or their dependencies in the future.

So, to work around this limitation, this change adds minimal skeleton versions of each package so that we can still run tests within travis-ci.

From the `.travis.yml` file, the `build` script is given a fake module path. And, the `build` script itself should remain unmodified so that local installations can still run unit tests with the true versions of `netifaces` and `netsnmp`.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-lab/collectd-mlab/41)
<!-- Reviewable:end -->

mtlynch commented:
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[fake-packages/netifaces.py, line 1 \[r1\]](https://reviewable.io:443/reviews/m-lab/collectd-mlab/41#-KGDzcsBbn41sC2vROVY:-KGDzcsCWwn2hbXFP3Bg:-546076001) ([raw file](https://github.com/m-lab/collectd-mlab/blob/02db7b52042300434fbac89419df65ec83a3891e/fake-packages/netifaces.py#L1)):*
Just because these files will probably be so unexpected to a reader without context, I think the bar for documentation is a bit high. Can we add a README that explains why we have to do this (including the limitations in Travis, bonus for including links) and then, in the module docstrings, direct the reader to the README?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/collectd-mlab/41)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
Review status: 2 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

*[fake-packages/netifaces.py, line 1 \[r1\]](https://reviewable.io:443/reviews/m-lab/collectd-mlab/41#-KGDzcsBbn41sC2vROVY:-KGEBI8X6OL5-fjmuw3c:1184467493) ([raw file](https://github.com/m-lab/collectd-mlab/blob/02db7b52042300434fbac89419df65ec83a3891e/fake-packages/netifaces.py#L1)):*
This is a good idea.

I've added fake-packages/README.md and links to the future location of the README under the `m-lab/collectd-mlab` repo in the package docstrings. Also added a comment in .travis.yml pointing to the same README.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/collectd-mlab/41)*
<!-- Sent from Reviewable.io -->

mtlynch commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 4 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[.travis.yml, line 19 \[r3\]](https://reviewable.io:443/reviews/m-lab/collectd-mlab/41#-KGEF4c4hXtgbZ_RWQft:-KGEF4c4hXtgbZ_RWQfu:909547024) ([raw file](https://github.com/m-lab/collectd-mlab/blob/8722f18b158134bfa316ca4576c010e8aea5706b/.travis.yml#L19)):*
(optional) I'd rather use relative paths than bake in an assumption that we're always going to be hosted on Github at this path. (ditto for rest of PR)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/collectd-mlab/41#-:-KGEFIRbkJZfdiXWJ3hI:1417088721)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
Review status: 3 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[.travis.yml, line 19 \[r3\]](https://reviewable.io:443/reviews/m-lab/collectd-mlab/41#-KGEF4c4hXtgbZ_RWQft:-KGE_drvNGPJf2O-qsD_:-622367394) ([raw file](https://github.com/m-lab/collectd-mlab/blob/8722f18b158134bfa316ca4576c010e8aea5706b/.travis.yml#L19)):*
Oh, okay. No problem. Updated to `fake-packages/README.md` throughout. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/collectd-mlab/41)*
<!-- Sent from Reviewable.io -->
